trunner/runners/common.py

```diff
@@ -305,26 +305,46 @@ def go(self):
         self.plo.send('go!\r\n')
 
 
-class Runner:
+class Runner(ABC):
     """Common interface for test runners"""
 
+    BUSY = 'BUSY'
+    SUCCESS = 'SUCCESS'
+    FAIL = 'FAIL'
+
+    def __init__(self):
+        # Busy status is set from the start to the end of the specified runner's run
+        self.status = Runner.BUSY
+        self.set_status(self.status)
+
+    def set_status(self, status):
+        """Method for sygnalising a current runner status: busy/failed/succeeded"""
+        # for now, not used in all target runners
+        self.status = status
+
+    @abstractmethod
     def flash(self):
         """Method used for flashing a device with the image containing tests."""
         pass
 
+    @abstractmethod
     def run(self, test):
         """Method used for running a single test case which is represented by TestCase class."""
         pass
 
 
 class DeviceRunner(Runner):
-    """This class provides interface to run test case using serial port"""
+    """This class provides interface to run tests on hardware targets using serial port"""
 
     def __init__(self, serial):
+        super().__init__()
         self.serial_port = serial[0]
         self.serial_baudrate = serial[1]
         self.serial = None
 
+    def set_status(self, status):
```

Comment:
not needed (function of the parent class will be called if it's not redefined in the child class)